src/test/java/com/amazon/opendistroforelasticsearch/jobscheduler/JobSchedulerSettingsTests.java

```diff
@@ -0,0 +1,119 @@
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.test.OpenSearchTestCase;
+
+@SuppressWarnings({"rawtypes"})
+public class JobSchedulerSettingsTests extends OpenSearchTestCase {
+
+    JobSchedulerPlugin plugin = null;
+    
+    @Before
+    public void setup() {
+        this.plugin = new JobSchedulerPlugin();
+    }
+
+    public void testAllLegacyOpenDistroSettingsReturned() {
+        List<Setting<?>> settings = plugin.getSettings();
+        assertTrue("legacy setting must be returned from settings", 
+            settings.containsAll(
+                Arrays.asList(
+                    LegacyOpenDistroJobSchedulerSettings.JITTER_LIMIT,
+                    LegacyOpenDistroJobSchedulerSettings.REQUEST_TIMEOUT,
+                    LegacyOpenDistroJobSchedulerSettings.SWEEP_BACKOFF_MILLIS,
+                    LegacyOpenDistroJobSchedulerSettings.SWEEP_BACKOFF_RETRY_COUNT,
+                    LegacyOpenDistroJobSchedulerSettings.SWEEP_PAGE_SIZE,
+                    LegacyOpenDistroJobSchedulerSettings.SWEEP_PERIOD
+                )
+            )
+        );
+    }
+
+    public void testAllOpenSearchSettingsReturned() {
+        List<Setting<?>> settings = plugin.getSettings();
+        assertTrue("legacy setting must be returned from settings", 
+            settings.containsAll(
+                Arrays.asList(
+                    JobSchedulerSettings.JITTER_LIMIT,
+                    JobSchedulerSettings.REQUEST_TIMEOUT,
+                    JobSchedulerSettings.SWEEP_BACKOFF_MILLIS,
+                    JobSchedulerSettings.SWEEP_BACKOFF_RETRY_COUNT,
+                    JobSchedulerSettings.SWEEP_PAGE_SIZE,
+                    JobSchedulerSettings.SWEEP_PERIOD
+                )
+            )
+        );
+    }
+
+    public void testLegacyOpenDistroSettingsFallback() {
+        assertEquals(
+            JobSchedulerSettings.REQUEST_TIMEOUT.get(Settings.EMPTY), 
+            LegacyOpenDistroJobSchedulerSettings.REQUEST_TIMEOUT.get(Settings.EMPTY)
+        );
+    }
+
+    public void testSettingsGetValue() {
+        Settings settings = Settings.builder().put("opensearch.jobscheduler.request_timeout", "42s").build();
+        assertEquals(JobSchedulerSettings.REQUEST_TIMEOUT.get(settings), TimeValue.timeValueSeconds(42)); 
+        assertEquals(LegacyOpenDistroJobSchedulerSettings.REQUEST_TIMEOUT.get(settings), TimeValue.timeValueSeconds(10)); 
+    }
+
+    public void testSettingsGetValueWithLegacyFallback() {
+        Settings settings = Settings.builder()
+            .put("opendistro.jobscheduler.request_timeout", "1s")
+            .put("opendistro.jobscheduler.sweeper.backoff_millis", "2ms")
+            .put("opendistro.jobscheduler.retry_count", 3)
+            .put("opendistro.jobscheduler.sweeper.period", "4s")
+            .put("opendistro.jobscheduler.sweeper.page_size", 5)
+            .put("opendistro.jobscheduler.jitter_limit", 6)
+        .build();
```

Comment:
Lets use LegacyOpenDistroJobSchedulerSettings constants here maybe

Reply:
I actually would rather not. If I were to make a mistake or a typo in the actual rename in the setting, or someone accidentally bulk replaces opendistro by openseaerch, the test will point to a behavior change or start failing. Which is what we really want. WDYT?